shared/qt/idian/components/ComboBox.cpp

```diff
@@ -0,0 +1,64 @@
+/******************************************************************************
+    Copyright (C) 2024 by Taylor Giampaolo <warchamp7@obsproject.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include <QTimer>
+
+#include <Idian/Row.hpp>
+
+#include <Idian/moc_ComboBox.cpp>
+
+using idian::ComboBox;
+
+ComboBox::ComboBox(QWidget *parent) : QComboBox(parent), Utils(this) {}
+
+void ComboBox::showPopup()
+{
+	if (allowOpeningPopup) {
+		QComboBox::showPopup();
+	}
+}
+
+void ComboBox::hidePopup()
+{
+	/* ToDo: Find a better way to do this
```

Comment:
`TODO` is usually automatically detected by IDEs and thus this line is potentially detected and highlighted in some way. In that case the actual to-do summary should be more expressive about what should be done.

Also is this a to-do to finish up Idian or will this stay as-is for the foreseeable future?

Reply:
I'll replace the `ToDo:` with something else. This is a Rodney comment but I didn't have a better solution either. As far as I'm concerned, this is a tad-bit hacky but fine to merge as-is.